src/js/themes/vanilla.js

```diff
@@ -195,6 +203,14 @@ export default {
     },
     weight: 300,
   },
+  icon: {
+    extend: css`
+      ${props => props.color && props.color !== 'plain' && `
+        fill: ${colorForName(props.color, props.theme)};
```

Comment:
This seems circular to me, that the theme would call a function referencing the theme.